lib/newlib/stubs.c

```diff
@@ -0,0 +1,131 @@
+//! \file
+//! \brief Newlib stubs implementation.
+//! \details Not all newlib functionality is allowed (by default) in theCore.
+//! Examples include forking or using memory allocations.
+
+#include <platform/execution.h>
+
+#include <stdlib.h>
+#include <sys/unistd.h>
+#include <sys/stat.h>
+#include <sys/times.h>
+#include <errno.h>
+
+#undef errno
+extern int errno;
+
+void _exit(int status)
+{
+    (void)status;
+    ecl_abort();
+}
+
+int _close(int file)
+{
+    (void)file;
+    return -1;
+}
+
+char *__env[1] = { 0 };
+
+char **environ = __env;
+
+int _execve(const char *name, char * const *argv, char * const * env)
+{
+    (void)name;
+    (void)argv;
+    (void)env;
+    errno = ENOMEM;
+    return -1;
+}
+
+int _fork(void)
+{
+    errno = EAGAIN;
+    return -1;
+}
+
+int _fstat(int file, struct stat *st)
+{
+    (void)file;
+    st->st_mode = S_IFCHR;
+    return 0;
+}
+
+int _getpid(void)
+{
+    return 1;
+}
+
+int _kill(int pid, int sig)
+{
+    (void)pid;
+    (void)sig;
+    errno = EINVAL;
+    return -1;
+}
+
+int _link(const char *old, const char *nw)
+{
+    (void)old;
+    (void)nw;
+    errno = EMLINK;
+    return -1;
+}
+
+off_t _lseek(int file, off_t oft, int dir)
+{
+    (void)file;
+    (void)oft;
+    (void)dir;
+    return 0;
+}
+
+int _read(int file, void *ptr, size_t len)
+{
+    (void)file;
+    (void)ptr;
+    (void)len;
+    return 0;
```

Comment:
How about returning `-1` and setting `errno` to `EINVAL` for all I/O operations? I find this nice as long as `open` is not supported since you can't retrieve a *valid* fd without opening anything